h2o-automl/src/test/java/ai/h2o/automl/targetencoding/BroadcastJoinTest.java

```diff
@@ -0,0 +1,115 @@
+package ai.h2o.automl.targetencoding;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.*;
+import water.fvec.*;
+
+import static org.junit.Assert.assertEquals;
+
+public class BroadcastJoinTest extends TestUtil {
+
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  private Frame fr = null;
+
+  @Test
+  public void joinPerformsWithoutLoosingOriginalOrderTest() {
+
+    Frame rightFr = null;
+    Vec emptyNumerator = null;
+    Vec emptyDenominator = null;
+    try {
+
+      fr = new TestFrameBuilder()
+              .withName("testFrame")
+              .withColNames("ColA", "fold")
+              .withVecTypes(Vec.T_CAT, Vec.T_NUM)
+              .withDataForCol(0, ar("a", "c", "b"))
+              .withDataForCol(1, ar(1, 0, 1))
+              .build();
+
+      rightFr = new TestFrameBuilder()
+              .withName("testFrame2")
+              .withColNames("ColA", "fold", "numerator", "denominator")
+              .withVecTypes(Vec.T_CAT, Vec.T_NUM, Vec.T_NUM, Vec.T_NUM)
+              .withDataForCol(0, ar("a", "b", "c"))
+              .withDataForCol(1, ar(1, 0, 0))
+              .withDataForCol(2, ar(22, 33, 42))
+              .withDataForCol(3, ar(44, 66, 84))
+              .build();
+
+      emptyNumerator = Vec.makeZero(fr.numRows());
+      fr.add("numerator", emptyNumerator);
+      emptyDenominator = Vec.makeZero(fr.numRows());
+      fr.add("denominator", emptyDenominator);
+      
+      Frame joined = BroadcastJoinForTargetEncoder.join(fr, new int[]{0}, 1, rightFr, new int[]{0}, 1);
+
+      Scope.enter();
+      assertStringVecEquals(cvec("a", "c", "b"), joined.vec("ColA"));
+      assertEquals(22, joined.vec("numerator").at(0), 1e-5);
+      assertEquals(42, joined.vec("numerator").at(1), 1e-5);
+      assertEquals(44, joined.vec("denominator").at(0), 1e-5);
+      assertEquals(84, joined.vec("denominator").at(1), 1e-5);
+      Scope.exit();
```

Comment:
aren't those assertions true also without fold columns?
I mean 3rd row should be the one with different behaviour, like null values, no?

Reply:
Yes you are right, but I was checking that I'm able to get what I expected to get(I was more concerned on the right values in each particular row). To make it more clear I will add extra checks, thank you!

Reply:
it is just difficult to create vec of ints with null in it....  inconvenient to check this kind of things